test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -3232,3 +3243,71 @@ UCS_TEST_P(test_ucp_sockaddr_protocols_err_sender,
 UCP_INSTANTIATE_CM_TEST_CASE(test_ucp_sockaddr_protocols_err_sender)
 UCP_INSTANTIATE_TEST_CASE_TLS_GPU_AWARE(test_ucp_sockaddr_protocols_err_sender,
                                         rc_no_ud, "rc_mlx5,rc_verbs")
+
+
+class test_ucp_sockaddr_iface_activate : public test_ucp_sockaddr {
+public:
+    static void
+    get_test_variants(std::vector<ucp_test_variant>& variants)
+    {
+        get_test_variants_mt(variants, UCP_FEATURE_TAG,
+                             CONN_REQ_TAG | TEST_MODIFIER_CM_USE_ALL_DEVICES,
+                             "tag");
+    }
+
+    virtual int is_skip_interface(const struct ifaddrs *ifa) const
+    {
+        return test_ucp_sockaddr::is_skip_interface(ifa) ||
+               ucs::is_rdmacm_netdev(ifa->ifa_name);
+    }
+
+    bool is_interface_activated(entity &e) const
+    {
+        ucp_worker_h worker = e.worker();
+        for (unsigned i = 0; i < worker->num_ifaces; ++i) {
+            if (ucp_worker_iface_is_activated(worker->ifaces[i])) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    void client_connect_disconnect()
+    {
+        /* Check state before connection establishment */
+        EXPECT_FALSE(is_interface_activated(receiver()));
+        int receiver_ep_count = receiver().get_num_eps();
+
+        /* Connect sender and receiver */
+        client_ep_connect_basic(get_ep_params(), 0, false);
+        wait_progress([&] {
+            return receiver().get_num_eps() == (receiver_ep_count + 1);
+        });
+
+        /* Check state after connection establishment */
+        EXPECT_EQ(0, sender().get_err_num());
+        EXPECT_TRUE(is_interface_activated(sender()));
+        EXPECT_TRUE(is_interface_activated(receiver()));
+
+        one_sided_disconnect(sender());
+        one_sided_disconnect(receiver(), 0, receiver_ep_count);
+        /* Wait for EP disconnect completion which might happen after
+         * one_sided_disconnect */
+        wait_progress([&] {
+            return !is_interface_activated(receiver());
+        });
```

Comment:
why need to wait for ! activated?
seems that after one_sided_disconnect the activation count should be already set to 0 because that function waits for disconnect to complete

Reply:
Yes, it was needed with one sided disconnect from the client side, but with disconnect on the receiver side it's redundant